systemtests/battery_test.go

```diff
@@ -2,78 +2,93 @@ package systemtests
-		volumes := genRandomVolumes(count)
+	for i := 0; i < totalIterations; i++ {
+		syncChan := make(chan struct{}, threadCount)
+		volumes := genRandomVolumes(threadCount)
 
-		for x := 0; x < count; x++ {
-			go func(volName string, x int) {
+		for workerID := 0; workerID < threadCount; workerID++ {
+			go func(volume string, workerID int) {
 				defer func() { syncChan <- struct{}{} }()
-				fqVolName := fqVolume("policy1", volName)
-
-				c.Assert(s.createVolume("mon0", fqVolName, nil), IsNil)
-				out, err := s.dockerRun("mon0", false, true, fqVolName, "sleep 10m")
-				c.Assert(err, IsNil, Commentf("Output: %s", out))
-				second, err := s.dockerRun("mon0", false, true, fqVolName, "sleep 10m")
-				log.Debug(second, err)
-				c.Assert(err, IsNil, Commentf("Output: %s", second))
-
+				fqVolName := fqVolume("policy1", volume)
+
+				c.Assert(s.createVolume("mon0", fqVolName, map[string]string{"unlocked": isUnlocked}), IsNil)
+				containerID, err := s.dockerRun("mon0", false, true, fqVolName, "sleep 10m")
+				c.Assert(err, IsNil, Commentf("Output: %s", containerID))
+
+				for x := 0; x < totalMounts; x++ {
+					id, err := s.dockerRun("mon0", false, true, fqVolName, "sleep 10m")
+					if unlocked, _ := strconv.ParseBool(isUnlocked); unlocked {
+						c.Assert(err, IsNil, Commentf("Output: %s", id))
+						dockerRmOut, err := s.mon0cmd(fmt.Sprintf("docker rm -f %s", strings.TrimSpace(id)))
+						if err != nil {
+							log.Error(strings.TrimSpace(dockerRmOut))
+						}
+						c.Assert(err, IsNil)
+					} else { // locked volumes
+						log.Debug("Volume %s already mounted in container %s", fqVolName, containerID)
+						c.Assert(err, NotNil)
+					}
+				}
 				if cephDriver() {
-					_, err = s.mon0cmd(fmt.Sprintf("mount | grep rbd | grep -q policy1.%s", volName))
+					_, err = s.mon0cmd(fmt.Sprintf("mount | grep rbd | grep -q %s", strings.Join([]string{"policy1", volume}, ".")))
 					c.Assert(err, IsNil)
-					out2, err := s.mon0cmd(fmt.Sprintf("docker exec %s ls /mnt", strings.TrimSpace(out)))
+					mountedDirContent, err := s.mon0cmd(fmt.Sprintf("docker exec %s ls /mnt", strings.TrimSpace(containerID)))
 					c.Assert(err, IsNil)
-					c.Assert(strings.TrimSpace(out2), Equals, "lost+found")
+					c.Assert(strings.TrimSpace(mountedDirContent), Equals, "lost+found")
 				}
 
-				out3, err := s.mon0cmd(fmt.Sprintf("docker rm -f %s", strings.TrimSpace(out)))
+				dockerRmOut, err := s.mon0cmd(fmt.Sprintf("docker rm -f %s", strings.TrimSpace(containerID)))
 				if err != nil {
-					log.Error(strings.TrimSpace(out3))
+					log.Error(strings.TrimSpace(dockerRmOut))
 				}
 				c.Assert(err, IsNil)
-			}(volumes[x], x)
+			}(volumes[workerID], workerID)
 		}
 
-		for x := 0; x < count; x++ {
+		for x := 0; x < threadCount; x++ {
 			<-syncChan
 		}
 
 		if cephDriver() {
-			_, err := s.mon0cmd("mount | grep -q rbd")
-			c.Assert(err, IsNil)
+			out, _ := s.mon0cmd("mount | grep -q rbd")
+			c.Assert(out, Equals, "")
```

Comment:
this will always succeed: because `-q` never prints anything, just returns an exit status.

Please revert this section to the left hand side, it was already correct.